src/Entity/PageManager.php

```diff
@@ -35,7 +37,7 @@ final class PageManager extends BaseEntityManager implements PageManagerInterfac
     public function __construct(
         string $class,
         ManagerRegistry $registry,
-        private SlugifyInterface $slugify,
+        private SlugifyInterface|SluggerInterface $slugger,
```

Comment:
You need to add a deprecation if the slugger passed is not instance of SluggerInterface.

And a next major comment to reduce the param to sluggerInterface.